pkg/flow/states/action-helpers.go

```diff
@@ -15,12 +15,15 @@ import (
 )
 
 type actionRetryInfo struct {
-	Children []ChildInfo
-	Idx      int
+	Children   []ChildInfo
+	Idx        int
+	Originator string
 }
 
 type actionResultPayload struct {
 	ActionID     string
+	Originator   string
```

Comment:
I don't think it's good to let the container report this information back to us. They should only report the action id, and we should look up this information ourselves.